rasa/core/policies/ensemble.py

```diff
@@ -34,53 +84,130 @@
     ActionExecutionRejected,
     ActionExecuted,
     DefinePrevUserUtteredFeaturization,
+    Event,
 )
-from rasa.core.exceptions import UnsupportedDialogueModelError
-from rasa.core.featurizers.tracker_featurizers import MaxHistoryTrackerFeaturizer
-from rasa.shared.nlu.interpreter import NaturalLanguageInterpreter
-from rasa.core.policies.policy import Policy, SupportedData, PolicyPrediction
-from rasa.core.policies.memoization import MemoizationPolicy, AugmentedMemoizationPolicy
-from rasa.core.policies.rule_policy import RulePolicy
-from rasa.core.training import training
 from rasa.shared.core.trackers import DialogueStateTracker
-from rasa.shared.core.generator import TrackerWithCachedStates
-from rasa.core import registry
-from rasa.utils.tensorflow.constants import EPOCHS
+import rasa.utils.io
 
 logger = logging.getLogger(__name__)
 
+# TODO: This is a workaround around until we have all components migrated to
+# `GraphComponent`.
+SimplePolicyEnsemble = SimplePolicyEnsemble
+PolicyEnsemble = PolicyEnsemble
+InvalidPolicyConfig = InvalidPolicyConfig
 
-class PolicyEnsemble:
-    versioned_packages = ["rasa", "tensorflow", "sklearn"]
 
-    def __init__(
-        self,
-        policies: List[Policy],
-        action_fingerprints: Optional[Dict[Any, Dict[Text, List]]] = None,
-    ) -> None:
-        self.policies = policies
-        self.date_trained = None
+def propagate_specification_of_rule_only_data(ensemble: List[Policy]) -> None:
+    """Extracts information on rule only data and propagates it to all policies.
 
-        self.action_fingerprints = action_fingerprints or {}
+    TODO: remove this once there's a corresponding new graph component taking
+    care of this
+    """
+    rule_policy = next(
+        (policy for policy in ensemble if isinstance(policy, RulePolicy)), None
+    )
+    if rule_policy:
+        rule_only_data = rule_policy.get_rule_only_data()
+    else:
+        rule_only_data = dict()
+    for policy in ensemble:
+        policy.set_shared_policy_states(rule_only_data=rule_only_data)
 
-        self._check_priorities()
-        self._check_for_important_policies()
 
-        self._set_rule_only_data()
+def validate_warn_rule_based_data_unused_or_missing(
```

Comment:
Mhm, actually we can't do this check as part of the `RulePolicy` as it wouldn't fire in case users don't add this. This would probably go into [this component](https://github.com/RasaHQ/rasa/issues/9336)

Reply:
(updated the issue description accordingly)